extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifier.java

```diff
@@ -0,0 +1,94 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.identityhub.verifier;
+
+import org.eclipse.dataspaceconnector.iam.did.spi.credentials.CredentialsVerifier;
+import org.eclipse.dataspaceconnector.iam.did.spi.document.DidDocument;
+import org.eclipse.dataspaceconnector.identityhub.client.IdentityHubClient;
+import org.eclipse.dataspaceconnector.identityhub.credentials.VerifiableCredentialsJwtUtils;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.result.AbstractResult;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Obtains and verifies credentials associated with a DID.
+ * The DID document contains an IdentityHub service, the IdentityHubCredentialsVerifier gets credentials from the
+ * IdentityHub instance and verifies the credentials.
+ */
+public class IdentityHubCredentialsVerifier implements CredentialsVerifier {
+
+    private static final String IDENTITY_HUB_SERVICE_TYPE = "IdentityHub";
+    private final IdentityHubClient identityHubClient;
+    private final Monitor monitor;
+    private final JwtCredentialsVerifier jwtCredentialsVerifier;
+    private final VerifiableCredentialsJwtUtils jwtPayloadParser;
+
+    /**
+     * Create a new credential verifier that uses an Identity Hub
+     *
+     * @param identityHubClient IdentityHubClient.
+     */
+    public IdentityHubCredentialsVerifier(IdentityHubClient identityHubClient, Monitor monitor, JwtCredentialsVerifier jwtCredentialsVerifier, VerifiableCredentialsJwtUtils JWTPayloadParser) {
```

Comment:
`JWTPayloadParser` should not start with a capital letter. Also why the choice of name?

Reply:
I have renamed it to `VerifiableCredentialsJWTService`, it will be extended in coming PRs with further methods to manipulate VC JWTs